src/collectors/jolokia/jolokia.py

```diff
@@ -119,53 +163,75 @@ def check_mbean(self, mbean):
         mbeanfix = self.clean_up(mbean)
         if self.config['regex'] is not None:
             for chkbean in self.mbeans:
-                if re.match(chkbean, mbean) is not None or \
-                   re.match(chkbean, mbeanfix) is not None:
+                if chkbean.match(mbean) is not None or \
+                   chkbean.match(mbeanfix) is not None:
                     return True
         else:
             if mbean in self.mbeans or mbeanfix in self.mbeans:
                 return True
 
     def collect(self):
-        listing = self.list_request()
-        try:
-            domains = listing['value'] if listing['status'] == 200 else {}
-            for domain in domains.keys():
-                if domain not in self.IGNORE_DOMAINS:
-                    obj = self.read_request(domain)
+        if not self.domains:
+            self._get_domains()
+        for domain in self.domains:
+            if domain not in self.IGNORE_DOMAINS:
+                obj = self.read_request(domain)
+                try:
                     mbeans = obj['value'] if obj['status'] == 200 else {}
-                    for k, v in mbeans.iteritems():
-                        if self.check_mbean(k):
-                            self.collect_bean(k, v)
-        except KeyError:
-            # The reponse was totally empty, or not an expected format
-            self.log.error('Unable to retrieve MBean listing.')
+                except KeyError:
+                    # The reponse was totally empty, or not an expected format
+                    self.log.error('Unable to retrieve domain %s.', domain)
+                    continue
+                for k, v in mbeans.iteritems():
+                    if self.check_mbean(k):
+                        self.collect_bean(k, v)
 
     def read_json(self, request):
         json_str = request.read()
         return json.loads(json_str)
 
-    def list_request(self):
+    def _list_request(self):
```

Comment:
If we're going to change this here, can we change the other "private" methods to use the "_" convention as well? We should just be consistent one way or another.